axelrod/strategies/ann.py

```diff
@@ -169,62 +172,40 @@ class EvolvedANN(ANN):
-                       -0.8053911061885284, -5363.722820739466, 171.58433274294117, -724.7468082647013,
-                       2500359.853524033, 1595.3955511798079, -4.254009123616706, -171.12968391407912,
-                       -32.30624102753424, -558.412338112568, -234.29754199019308, -18768.34057250429,
-                       8338.792126484348, -0.18593140210730602, -7.758804964874875, 0.39736677884665267,
-                       547.0567585452197, 1.1969366369973133, 0.4861465741177498, -51.19319208716985,
-                       12.775051406025534, -0.09185362260212569, 22.08417300332754, -5090.013231748707,
-                       -0.814394991797045, 1.1534025840023847, 8.390439959276764, -0.02227253403481858,
-                       0.14162040507921927, -0.011508263843203926, 0.22372493104861083, 0.7754713610627112,
-                       0.1044033140236981, -4.377055307648915, -41.898221495326574, -18656.755601828827,
-                       -134.56719406539244, -2405.8148785743474, 16864.049985157206, -0.5124682025216784,
-                       14521.069005125159, -10.740782200739309, 18756.807715014013, -1723.9353962656946,
-                       87029.99828299093, 5.7383786020894195e-05, 4762.960401619296, 0.7331769713238158,
-                       -308.5673034493341, 85.29725765515369, 0.4268843538235295, -0.17788805472511407,
-                       -1.1727033611646802, 7578.6822604990175, 0.5124673187864222, 0.1595627909684813,
-                       -145.93742731401096, -2954.234440189563, 0.009672881359732015, 106.4646644917487,
-                       -0.050606976105730346, 2.3904047264403596, -4.987645640997455, -43.22984692765006,
-                       -36.177108409134966, -0.3812547430698569, -2959.4921368963633, -1.8635802741029985,
-                       0.020513128847167047, -0.9179124323385958]
-
-        (i2h, h2o, bias) = split_weights(
-            weights,
-            input_values,
-            hidden_layer_size
-        )
-        ANN.__init__(self, i2h, h2o, bias)
+        num_features, num_hidden, weights = nn_weights['1']
+        ANN.__init__(self, weights, num_features, num_hidden)
+
+
+class EvolvedANN5(ANN):
+    """
+    A strategy based on a pre-trained neural network with 17 features and a
+    hidden layer of size 5.
+
+    Names:
+
+     - EvolvedANN5: : Original name by Marc Harper.
+    """
+
+    name = "EvolvedANN5"
+
+    @init_args
+    def __init__(self):
+        num_features, num_hidden, weights = nn_weights['5']
+        ANN.__init__(self, weights, num_features, num_hidden)
+
+
+class EvolvedANNNoise05(ANN):
+    """
+    A strategy based on a pre-trained neural network, trained with noise=0.05.
+
+    Names:
+
+     - EvolvedANNNoise05: Original name by Marc Harper.
+    """
+
+    name = "EvolvedANNNoise05"
+
+    @init_args
+    def __init__(self):
+        num_features, num_hidden, weights = nn_weights['05']
```

Comment:
My comment about the readability of the key (in the dataset) holds here too. But perhaps make it even more verbose than a single number. Could we perhaps have `10layers-noise-05` for here? And `10layers` for above? (Or similar).